src/components/CustomerAppeal/CustomerAppealScreen.tsx

```diff
@@ -81,41 +81,68 @@ export const CustomerAppealScreen: FunctionComponent<NavigationProps> = ({
   const { allQuotaResponse } = useQuota(ids, sessionToken, endpoint);
 
   const { i18nt } = useTranslate();
-
   const getReasons = (): Reason[] => {
     return transform(
       allProducts ?? [],
       (result: Reason[], policy) => {
         if (policy.categoryType === "APPEAL") {
           const policyLimit = policy.quantity.limit;
+          const policyThreshold = policy.alert?.threshold;
           const quotaResponse = allQuotaResponse?.remainingQuota.find(
-            (quota) => quota.category === policy.category
+            (quota) =>
+              quota.category === policyThreshold ||
+              quota.category === policy.category
           );
           let descriptionAlert: string | undefined = undefined;
-          if (
-            quotaResponse &&
-            policy.alert &&
-            policyLimit - quotaResponse.quantity >= policy.alert.threshold
-          ) {
-            descriptionAlert = policy.alert.label;
+          if (typeof policyThreshold === "number") {
+            if (
+              quotaResponse &&
+              policy.alert &&
+              policyLimit - quotaResponse.quantity >= policyThreshold
+            ) {
+              descriptionAlert = policy.alert.label;
+            }
+            result.push({
+              category: policy.category,
+              description: policy.name,
+              descriptionAlert,
+            });
+            // policy threshold is string when its visibility is conditional to other policy that is
+            // written on the threshold
```

Comment:
We can move this down into the `else if` block.

We might want to state some of our assumptions too, such as:

- Policies with dependencies on other policies should be defined after their dependents
- Policies with dependencies show up depending on whether its dependent policy's `descriptionAlert` is shown